backend/engine/plugins/owasp_dependency_check/main.py

```diff
@@ -101,15 +114,28 @@ def parse_scanner_output_json(repo_path, returncode):
     """
     results = []
     success = returncode == 0
+    errors = []
 
     if not os.path.exists(f"{repo_path}dependency-check-report.json"):
         return {"output": [], "errors": ["No report file found"], "success": success}
 
     with open(f"{repo_path}dependency-check-report.json") as json_file:
         data = json.load(json_file)
+        results = parse_vulnerabilities(data)
+        errors = parse_errors(data)
+
+    success = success & len(results) == 0
+    return {"output": results, "errors": errors, "success": success}
+
+
+def parse_vulnerabilities(data):
```

Comment:
if possible, insert expected data type for parameter and expected data type to be returned to make types more strict.